data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3InputStream.java

```diff
@@ -598,4 +609,12 @@ static int readFullyDirectBuffer(InputStream f, ByteBuffer buf, byte[] temp) thr
 
         return totalBytesRead;
     }
+
+    private void recordS3Exception(final S3Exception ex) {
+        if (ex.statusCode() == HttpStatusCode.NOT_FOUND) {
+            s3ObjectPluginMetrics.getS3ObjectsFailedNotFoundCounter().increment();
+        } else if (ex.statusCode() == HttpStatusCode.FORBIDDEN) {
+            s3ObjectPluginMetrics.getS3ObjectsFailedAccessDeniedCounter().increment();
+        }
```

Comment:
How about counter for other status code ?

Reply:
This change just maintains the existing behavior. There probably should be additional status codes checked but that's outside the scope of this change.